sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -3758,7 +3758,10 @@ case class ArrayUnion(left: Expression, right: Expression) extends ArraySetLike
       } else {
         val arrayUnion = classOf[ArrayUnion].getName
         val et = ctx.addReferenceObj("elementTypeUnion", elementType)
-        val order = ctx.addReferenceObj("orderingUnion", ordering)
+        // Some data types (e.g., `BinaryType`) have anonymous classes for ordering and
+        // `getCanonicalName` returns null in these classes. Therefore, we need to
+        // explicitly set `className` here.
```

Comment:
nit: as we are adding this comment, shall we also mention that Janino works anyway, but JDK complains here?